Ensure that no focusable elements are marked as aria-hidden or given a role of none/presentation

To ensure that elements remain understandable and operable, user agents will ignore `aria-hidden="true"`
or a role of none/presentation on a focusable element.

Even if a element has a `tabindex` of `-1`, it can still be theoretically focused and may appear in
element lists generated by assistive technologies such as a list of links.

Focusable elements include form controls, elements with a `tabindex`, links, `<audio>`, `<video>` and
elements with a `draggable` attribute and `<iframe>`s.

It also includes elements with overflow scrolling; however they are not currently supported by this test.

Incorrect code for this rule:
```html
  <input aria-hidden="true" />
  <div aria-hidden="true"><button>Submit</button></div>
  <div tabindex="-1" role="presentation" />
```

### Related specifications

* https://www.w3.org/TR/html52/editing.html#focusable
* https://www.w3.org/TR/wai-aria-1.1/#presentation
* https://www.w3.org/TR/wai-aria-1.1/#aria-hidden
* https://w3c.github.io/using-aria/

